src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+using Internal.IO;
+
+namespace System
+{
+    internal class StartupHookProvider
+    {
+        private static string initializeMethodName = "Initialize";
+
+        // Parse a string specifying a list of assemblies and types
+        // containing a startup hook, and call each hook in turn.
+        private static void ProcessStartupHooks(IntPtr startupHooksVariable)
+        {
+            Debug.Assert(startupHooksVariable != IntPtr.Zero);
+            string[] startupHooks = Marshal.PtrToStringUTF8(startupHooksVariable).Split(new char[] { Path.PathSeparator });
```

Comment:
Use Split overload that takes `char` argument to avoid allocating a single element array.

Reply:
This will go away with the next commit.

Reply:
Still here

Reply:
Sorry, missed this earlier but it's gone now.